### ISSUE TITLE
How To: Config ShredOS to boot and automatically invoke clean command

### ISSUE THREAD

**Issue by dgdomtar** [PartialVolume/shredos.x86_64#279]
We are decomming devices to be given to a non profit. All we need to do essentially is run the Diskpart clean command to wipe the drives.

Can ShredOS be setup to automatically do this once it's booted via USB? IE no interaction?

**Comment by PartialVolume** [comment]
Yes, you can create a auto nuke device by editing the grub.cfg files and adding nwipe_options=" --autonuke". See https://github.com/PartialVolume/shredos.x86_64?tab=readme-ov-file#how-to-change-the-default-nwipe-options-so-the-change-persists-between-reboots

However, it obviously won't be using DiskPart which is Windows. I couldn't figure out what sort of wipe DiskPart clean is doing. Traditional block writing or ATA secure erase? Any Windows users know?

**Comment by gorbiWTF** [comment]
https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/clean
> On master boot record (MBR) disks, only the MBR partitioning information and hidden sector information is overwritten.
> 
> On GUID Partition Table (gpt) disks, the gpt partitioning information, including the Protective MBR, is overwritten. There is no hidden sector information.

A for effort, I guess...

**Comment by PartialVolume** [comment]
> https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/clean
> 
> > On master boot record (MBR) disks, only the MBR partitioning information and hidden sector information is overwritten.
> > On GUID Partition Table (gpt) disks, the gpt partitioning information, including the Protective MBR, is overwritten. There is no hidden sector information.
> 
> A for effort, I guess...

I guess the clean command is not really for forensic destruction of data. Seems like the bare minimum it can get away with.  With a few forensic recovery tools a lot of the data could be recovered quite easily. It does make you wonder how many  unsuspecting people use clean thinking they deleted their data.

**Comment by githubcom2006** [comment]
To achieve 100% full automation with zero human interaction—where plugging in the USB and booting the machine instantly triggers a wipe without clicking or pressing a single key—you can utilize the hidden --autonuke parameter in [[ShredOS GitHub Repository](https://github.com/PartialVolume/shredos.x86_64)](https://github.com/PartialVolume/shredos.x86_64). [1] 
When configured this way, the USB will bypass all GUI confirmations, automatically detect every connected target storage drive, and begin zero-filling them immediately upon boot. [1, 2] 
## Step 1: Flash ShredOS to Your USB

   1. Download the latest version of ShredOS (the .img or .iso file).
   2. Insert your USB flash drive and use [[Rufus](https://rufus.ie/)](https://rufus.ie/) to write the image to the USB. [3, 4] 

## Step 2: Enable the "Autonuke" Override
To completely remove the manual safety confirmation, you must append the --autonuke and --nogui flags directly to the kernel command line parameters. [1, 2] 

   1. Safely eject and re-insert the USB drive into your current working computer to view its file structure.
   2. Open the partition named "SHREDOS" or the main USB folder.
   3. Locate the bootloader configuration file. Modern computers use UEFI, so look inside the boot or EFI directory for a file named grub.cfg.
   4. Right-click grub.cfg and select Open with Notepad (or any text editor). [1, 3] 

## Step 3: Modify the GRUB Configurations
Find the default menu entry line inside the file. It usually looks similar to this:

linux /boot/vmlinuz nwipe_options=""

Modify the nwipe_options section to include --autonuke, --nogui, and your preferred method. For a completely automated zero-fill, change that exact line to look like this: [1, 2] 

linux /boot/vmlinuz nwipe_options="--autonuke --nogui --method=zero"


* --autonuke: Instructs the engine to completely skip the interface screen, select all visible storage devices, and immediately execute the wipe with zero warnings or confirmation checks.
* --nogui: Suppresses the graphical user interface completely, running everything automatically via the back-end command line terminal.
* --method=zero: Sets the default wiping pattern to a high-speed zero-fill pass. [1, 2, 5] 

Save the file and close your text editor.
------------------------------
## Step 4: The Zero-Interaction Wiping Process

   1. Plug the customized USB drive into the target computer.
   2. Turn the computer on and force it to boot from the USB using the device's boot hotkey (e.g., F12, F11, F8, or Esc).
   3. The moment the computer selects the USB, ShredOS will initialize, run the command string, and immediately begin wiping every connected hard drive and SSD down to bare metal—without requiring you to touch the mouse or keyboard again. [2, 3, 6, 7] 

## ⚠️ Critical Safety Risks

* Weaponized Drive: Once modified, this USB drive acts as a live, automated data-destruction tool. If you accidentally boot your own daily-use computer with this USB left plugged into a slot, it will destroy your operating system and files instantly without asking for permission. [8] 
* Clear Labeling: It is highly recommended to physically label this USB drive with bright red tape or marker so it is never mixed up with standard storage media.


[1] [[https://github.com](https://github.com/PartialVolume/shredos.x86_64/issues/279)](https://github.com/PartialVolume/shredos.x86_64/issues/279)
[2] [[https://grokipedia.com](https://grokipedia.com/page/Nwipe)](https://grokipedia.com/page/Nwipe)
[3] [[https://www.youtube.com](https://www.youtube.com/watch?v=3Cl-LtK39rs&vl=en)](https://www.youtube.com/watch?v=3Cl-LtK39rs&vl=en)
[4] [[https://www.youtube.com](https://www.youtube.com/watch?v=fSPBlMdBBrc&t=1)](https://www.youtube.com/watch?v=fSPBlMdBBrc&t=1)
[5] [[https://www.youtube.com](https://www.youtube.com/watch?v=ONy3Sq-msa0)](https://www.youtube.com/watch?v=ONy3Sq-msa0)
[6] https://shredos.org/
[7] [[https://medium.com](https://medium.com/@neovise/6-best-free-automation-tools-to-boost-productivity-2025-aefbcda38677)](https://medium.com/@neovise/6-best-free-automation-tools-to-boost-productivity-2025-aefbcda38677)
[8] [[https://www.reddit.com](https://www.reddit.com/r/sysadmin/comments/fxgxkt/noninteractive_dban_usb_flashdrive_method/)](https://www.reddit.com/r/sysadmin/comments/fxgxkt/noninteractive_dban_usb_flashdrive_method/)
